google/cloud/storage/internal/xml_builders.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/xml_builders.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+#include "google/cloud/internal/make_status.h"
+
+namespace google {
+namespace cloud {
+namespace storage_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+namespace {
+
+auto constexpr kMaxPartNumber = 10000U;
```

Comment:
`constexpr` globals already have internal linkage, so the anonymous namespace is redundant in practice.  That said, perhaps this value should somehow be part of the interface.  It sounds like something the caller would want to know.

That said, this doesn't look like something that should be enforced on the client side.

Reply:
Funnily, I asked to move this to the `.cc` file.  I agree this is not something the library should enforce:

https://github.com/googleapis/google-cloud-cpp/blob/main/doc/adr/2018-06-19-do-not-duplicate-server-side-validation.md

Unless you need this for the library to work correct, just remove it.